src/Orleans/Streams/PersistentStreams/StreamQueueBalancerType.cs

```diff
@@ -39,5 +39,10 @@ public enum StreamQueueBalancerType
         /// That is, it does not rebalance queues based on dymanic changes in the cluster Membership.
         /// </summary>
         StaticClusterConfigDeploymentBalancer,
+
+        /// <summary>
+        /// Use user configured custom queue balancer
+        /// </summary>
+        CustomBalancer
```

Comment:
It would result in a larger change, but I have strong preference to fetching the balancer from DI and only using this type if no balancer has been added to the container. The NamedServices DI feature can hopefully be used so that different providers can have different balancers.